api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -171,14 +154,17 @@ public Patient registerPatient(Patient patient, List<Relationship> relationships
 				} else {
 					throw new APIException("Only one side of a relationship should be specified");
 				}
-				
+
 				personService.saveRelationship(relationship);
 			}
 		}
-		
+
 		EventMessage eventMessage = new EventMessage();
 		eventMessage.put(RegistrationCoreConstants.KEY_PATIENT_UUID, patient.getUuid());
+
+		//TODO incorrect set of "UUID" each time it generates new RANDOM value, instead of set uuid of creator.
 		eventMessage.put(RegistrationCoreConstants.KEY_REGISTERER_UUID, patient.getCreator().getUuid());
+		eventMessage.put(RegistrationCoreConstants.KEY_REGISTERER_ID, String.valueOf(patient.getCreator().getId()));
```

Comment:
@djazayeri because UUID is incorrect, and I can't figure out how to set it correctly.